conjure_oxide/src/unstable/solver_interface/adaptors/minion.rs

```diff
@@ -0,0 +1,306 @@
+            str_name,
+            ranges.len()
+        )));
+    }
+
+    let range = ranges.first().ok_or(ModelInvalid(format!(
+        "variable {:?} has no range",
+        str_name
+    )))?;
+
+    let (low, high) = match range {
+        conjureast::Range::Bounded(x, y) => Ok((x.to_owned(), y.to_owned())),
+        conjureast::Range::Single(x) => Ok((x.to_owned(), x.to_owned())),
+        #[allow(unreachable_patterns)]
+        x => Err(ModelFeatureNotSupported(format!("{:?}", x))),
+    }?;
+
+    _try_add_var(
+        str_name.to_owned(),
+        minionast::VarDomain::Bound(low, high),
+        minion_model,
+    )
+}
+
+fn _parse_booldomain_var(
+    name: &conjureast::Name,
+    minion_model: &mut minionast::Model,
+) -> Result<(), SolverError> {
+    let str_name = _name_to_string(name.to_owned());
+    _try_add_var(
+        str_name.to_owned(),
+        minionast::VarDomain::Bool,
+        minion_model,
+    )
+}
+
+fn _try_add_var(
+    name: minionast::VarName,
+    domain: minionast::VarDomain,
+    minion_model: &mut minionast::Model,
+) -> Result<(), SolverError> {
+    minion_model
+        .named_variables
+        .add_var(name.clone(), domain)
+        .ok_or(ModelInvalid(format!(
+            "variable {:?} is defined twice",
+            name
+        )))
+}
+
+fn parse_exprs(
+    conjure_model: &conjureast::Model,
+    minion_model: &mut minionast::Model,
+) -> Result<(), SolverError> {
+    for expr in conjure_model.get_constraints_vec().iter() {
+        parse_expr(expr.to_owned(), minion_model)?;
+    }
+    Ok(())
+}
+
+fn parse_expr(
+    expr: conjureast::Expression,
+    minion_model: &mut minionast::Model,
+) -> Result<(), SolverError> {
+    match expr {
+        conjureast::Expression::SumLeq(_metadata, lhs, rhs) => {
+            minion_model.constraints.push(minionast::Constraint::SumLeq(
+                read_vars(lhs)?,
+                read_var(*rhs)?,
+            ));
+            Ok(())
+        }
+        conjureast::Expression::SumGeq(_metadata, lhs, rhs) => {
+            minion_model.constraints.push(minionast::Constraint::SumGeq(
+                read_vars(lhs)?,
+                read_var(*rhs)?,
+            ));
+            Ok(())
+        }
+        conjureast::Expression::Ineq(_metadata, a, b, c) => {
+            minion_model.constraints.push(minionast::Constraint::Ineq(
+                read_var(*a)?,
+                read_var(*b)?,
+                minionast::Constant::Integer(read_const(*c)?),
+            ));
+            Ok(())
+        }
+        conjureast::Expression::Neq(_metadata, a, b) => {
+            minion_model
+                .constraints
+                .push(minionast::Constraint::WatchNeq(
+                    read_var(*a)?,
+                    read_var(*b)?,
+                ));
+            Ok(())
+        }
+        x => Err(ModelFeatureNotSupported(format!("{:?}", x))),
+    }
```

Comment:
This is fine for now, but long term this should probably be autogenerated, or done dynamically (minion generates this as a big JSON file. It would be easy to make Minion output that JSON file, I don't think it's currently stored somewhere nciely.